src/pfe/file-watcher/server/test/functional-test/suites/generic/tests/project.ts

```diff
@@ -0,0 +1,64 @@
+/*******************************************************************************
+ * Copyright (c) 2019 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v20.html
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+import mocha from "mocha";
+import { expect } from "chai";
+import * as _ from "lodash";
+import path from "path";
+
+import * as genericLib from "../../../lib/generic";
+
+import * as app_configs from "../../../configs/app.config";
+
+export default class ProjectTest {
+    testName: string;
+
+    constructor(testName: string) {
+        this.testName = testName;
+    }
+
+    run(runOnly?: boolean): void {
+        (runOnly ? describe.only : describe)(this.testName, () => {
+            this.runGetProjectTypes();
+        });
+    }
+
+    private runGetProjectTypes(): void {
+        describe("getProjectTypes function", () => {
+            it("get project type without a location", async () => {
+                const info: any = await genericLib.getProjectTypes();
+                expect(info);
+                expect(info.statusCode);
+                expect(info.statusCode).to.equal(200);
+                expect(info.types);
+                expect(info.types).to.deep.equal(["liberty", "spring", "swift", "nodejs", "docker"]); // later just check it with the app config list of project types
+            });
+
+            it("get project type with a defined project location", async () => {
+                const projectType = "spring";
+                const info: any = await genericLib.getProjectTypes(path.join(app_configs.codewindWorkspaceDir, app_configs.projectPrefix + projectType));
+                expect(info);
+                expect(info.statusCode);
+                expect(info.statusCode).to.equal(200);
+                expect(info.types);
+                expect(info.types).to.deep.equal([projectType]);
+            });
+
+            it("get project type with an undefined project location", async () => {
```

Comment:
I run the tests, one test case failed, issue is we expect [] but actually it returns ['docker'] for test case `get project type with an undefined project location`

![Screen Shot 2019-08-12 at 3 04 12 PM](https://user-images.githubusercontent.com/15131537/62890948-8a5e2f00-bd12-11e9-9991-a0fc6a0fdb01.png)

Reply:
I fixed this. It should work now